pkg/app/appserver/proc_manager.go

```diff
@@ -58,24 +60,18 @@ func (m *procManager) Run(log *logging.Logger, c appcommon.Config, args []string
 		return 0, err
 	}
 
-	if err := p.Run(); err != nil {
+	if err := m.rpcServer.Register(p.key); err != nil {
 		return 0, err
 	}
 
 	m.mx.Lock()
-	if _, ok := m.procs[c.Name]; ok {
```

Comment:
We still need this check to ensure that we won't run any app duplicates

Reply:
@Darkren It's checked by `m.Exists(c.Name)`. Or do you mean something different?

Reply:
Yep, sorry, probably missed this part